src/core.c/Sequence.pm6

```diff
@@ -38,9 +43,9 @@ nqp::p6configposbindfailover(Positional, PositionalBindFailover); # Binder
 Routine.'!configure_positional_bind_failover'(Positional, PositionalBindFailover); # Multi-dispatch
 
 my role Sequence does PositionalBindFailover {
-    multi method Array(::?CLASS:D:) { Array.from-iterator(self.iterator) }
-    multi method List(::?CLASS:D:)  { List.from-iterator(self.iterator) }
-    multi method Slip(::?CLASS:D:)  { Slip.from-iterator(self.iterator) }
+    multi method Array(::?CLASS:D:) { self.list.Array }
```

Comment:
I think in the case there is no cached list already, this would double roughly memory use, since it will need to reify the elements in a `List` as well as put them into `Scalar` containers in the `Array`?

Reply:
(Probably this should be done conditional on there being a cached list, and otherwise just do what was done before.)

Reply:
You may have a point there. Let me review both the lazy and non-lazy cases.

Reply:
I think you're right. I'll revert that change as `self.iterator` will be based on the cached list if it exists.